Libs/MRML/Core/vtkMRMLSegmentationStorageNode.cxx

```diff
@@ -923,6 +923,11 @@ int vtkMRMLSegmentationStorageNode::ReadPolyDataRepresentation(vtkMRMLSegmentati
     {
     // Get poly data representation
     vtkPolyData* currentPolyData = vtkPolyData::SafeDownCast(multiBlockDataset->GetBlock(blockIndex));
+    if (!currentPolyData)
+      {
+      vtkErrorMacro("ReadPolyDataRepresentation: Could not read block " << blockIndex);
```

Comment:
If we can access the segment id or name then it could be nice to include that in the log message.

Reply:
Unfortunately, we don't have the name or id. They are normally read from the FieldData of the polydata.

Reply:
OK, then this is really the best we can do at this point.